src/Catty/ViewController/Upload/UploadViewController.swift

```diff
@@ -259,171 +205,44 @@ class UploadViewController: UIViewController {
-                                 forHTTPBody: &body)
-
-            //Project Description
-            setFormDataParameter(projectDescriptionTag,
-                                 with: project?.header.programDescription.data(using: .utf8),
-                                 forHTTPBody: &body)
-
-            //User Email
-            if UserDefaults.standard.value(forKey: kcEmail) != nil {
-                setFormDataParameter(userEmailTag,
-                                     with: (UserDefaults.standard.value(forKey: kcEmail) as! String).data(using: .utf8),
-                                     forHTTPBody: &body)
-            }
-
-            //checksum
-            setFormDataParameter(fileChecksumParameterTag,
-                                 with: checksum?.data(using: .utf8),
-                                 forHTTPBody: &body)
-
-            //token
-            let token = JNKeychain.loadValue(forKey: kUserLoginToken) as! String
-            setFormDataParameter(tokenParameterTag,
-                                 with: token.data(using: .utf8),
-                                 forHTTPBody: &body)
-
-            //Username
-            setFormDataParameter(userNameTag,
-                                 with: (UserDefaults.standard.value(forKey: kcUsername) as! String).data(using: .utf8),
-                                 forHTTPBody: &body)
-
-            //Language
-            setFormDataParameter(deviceLanguageTag,
-                                 with: NSLocale.preferredLanguages[0].data(using: .utf8),
-                                 forHTTPBody: &body)
-
-            //zip file
-            setAttachmentParameter(uploadParameterTag,
-                                   with: zipFileData,
-                                   forHTTPBody: &body)
-
-            // close request form
-            if let anEncoding = "--\(httpBoundary)--\r\n".data(using: .utf8) {
-                body.append(anEncoding)
-            }
-            // set request body
-            request.httpBody = body
-
-            let postLength = String(format: "%lu", UInt(body.count))
-            request.addValue(postLength, forHTTPHeaderField: "Content-Length")
-
-            // debug output
-            let string1 = String(data: request.httpBody!,
-                                 encoding: String.Encoding.utf8) ?? "Data could not be printed"
-            debugPrint(string1)
-
-            // start request
-            let dataTask = session!.dataTask(with: request as URLRequest) { data, _, error in
+        let fileManager = CBFileManager()
+        let uploader = StoreProjectUploader(fileManager: fileManager)
+        if let project = project {
+            uploader.upload(project: project,
+                            completion: { error in
                 self.enableUploadView()
-                if error != nil {
-                    debugPrint("Connection could not be established")
+                guard let error = error else {
```

Comment:
Please change to `if let` or similar so that the success path is the default path being executed and the "error-path" is the path executed within the if-statement.